oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUtils.java

```diff
@@ -281,4 +287,35 @@ public static String getCaller(@Nullable String[] ignoredJavaPackages) {
         // if every element is ignored, we assume it's an internal request
         return "(internal)";
     }
+
+    /**
+     * Returns a Map consisting of checkpoints and checkpointInfoMap filtered based on a given predicate
+     * which can utilise checkpoint info map for filtering
+     * @param store
+     * @param predicate predicate used for filtering of checkpoints
+     * @return Map<String, Map<String,String>> filteredCheckpoint Map
+     */
+    public static Map<String, Map<String,String>> getFilteredCheckpoints(NodeStore store, Predicate<Map.Entry<String, Map<String,String>>> predicate) {
+        return StreamSupport.stream(store.checkpoints().spliterator(), false)
+                .collect(Collectors.toMap(str -> str,
+                        store::checkpointInfo))
+                .entrySet()
+                .stream()
+                .filter(predicate)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    }
+
+    /**
+     * Returns a map with checkpoint name as key and checkpoint metadata map as value, sorted on the basis of particular key in the metadata map.
```

Comment:
could you rewrite what this could sort based on? Imo: `sorted on the basis of particular key in the metadata map` isn't super clear. And an example would be very helpful.

Reply:
>sorted on the basis of particular key in the metadata map.

every checkpoint has an associated metadata map and this sort can be based on any of the keys of the metadata map. 

Added an example to try and make it more clear.